gui/components/tracker_entrance_label.py

```diff
@@ -74,3 +89,124 @@ def mouseReleaseEvent(self, ev: QMouseEvent) -> None:
+                        req, self.recent_search, TOD.ALL, self.world
+                    )
+                    else "red"
+                )
+                # Get a pretty name for the item if it is the first stage of a progressive item
+                name = pretty_name(req.args[0].name, 1)
+                return f'<span style="color:{color}">{name}</span>'
+            case RequirementType.COUNT:
+                # Determine if the user has enough of this item marked
+                color = (
+                    "dodgerblue"
+                    if evaluate_requirement_at_time(
+                        req, self.recent_search, TOD.ALL, self.world
+                    )
+                    else "red"
+                )
+                # Get a pretty name for the progressive item
+                name = pretty_name(req.args[1].name, req.args[0])
+                return f'<span style="color:{color}">{name}</span>'
+            case RequirementType.WALLET_CAPACITY:
+                # Determine if the user has enough wallet capacity for this requirement
+                color = (
+                    "dodgerblue"
+                    if evaluate_requirement_at_time(
+                        req, self.recent_search, TOD.ALL, self.world
+                    )
+                    else "red"
+                )
+                # TODO: Properly expand into wallet combinations
+                return f'<span style="color:{color}">Wallet >= {req.args[0]}</span>'
+            case RequirementType.GRATITUDE_CRYSTALS:
+                # Determine if the user has enough gratitude crystals marked
+                color = (
+                    "dodgerblue"
+                    if evaluate_requirement_at_time(
+                        req, self.recent_search, TOD.ALL, self.world
+                    )
+                    else "red"
+                )
+                return f'<span style="color:{color}">{req.args[0]} Gratitude Crystals</span>'
+            case RequirementType.OR:
+                # Recursively join requirements with "or"
+                # Only include parentheses if not at the top level (where they'd be redundant)
+                return (
+                    ("" if is_top_level else "(")
+                    + " or ".join([self.format_requirement(a, False) for a in req.args])
+                    + ("" if is_top_level else ")")
+                )
+            case RequirementType.AND:
+                # Recursively join requirements with "and"
+                # Only include parentheses if not at the top level (where they'd be redundant)
+                return (
+                    ("" if is_top_level else "(")
+                    + " and ".join(
+                        [self.format_requirement(a, False) for a in req.args]
+                    )
+                    + ("" if is_top_level else ")")
+                )
+            case _:
+                raise ValueError("unreachable")
```

Comment:
This code doesn't look meaningfully different from the code in `tracker_location_label`, it's probably possible to deduplicate that too?

Reply:
These have now been moved to tooltips.py, accepting the relevant label and requirement as arguments (though the tracker label argument can't be type annotated due to a circular import).